cylc/flow/network/graphql.py

```diff
@@ -80,6 +78,17 @@ def grow_tree(tree, path, leaves=None):
         tree_loc[len(path) % 2].update({'leaves': leaves})
 
 
+def execution_result_to_dict(execution_result):
+    result = {}
+    if execution_result.data:
+        result["data"] = execution_result.data
+    if execution_result.errors:
+        result["errors"] = [
+            error.formatted for error in execution_result.errors
+        ]
+    return result
```

Comment:
@MetRonnie, please could you check the data/errors fields in graphQL output are as expected.

Reply:
This function is only used in `cylc.flow.network.server.WorkflowRuntimeServer.graphql()`. It seems to me that we could just make use of the execution result object directly there

Reply:
Used here too

https://github.com/cylc/cylc-uiserver/pull/672/files#diff-f3b703a693bfd4b3fc21d0f3801d739bf17d4eaf193004628d8151b7995c75d2R343

Reply:
Ah right, I will take a closer look

Reply:
We should be able to just use `ExecutionResult.formatted` instead of this.

See https://github.com/graphql-python/graphql-core/blob/b7935f75511dde280325631541f4a8a24b245709/src/graphql/execution/execute.py#L141-L149